src/modules/hip/kernel/tensor_max.hpp

```diff
@@ -0,0 +1,410 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 - Reduction Stage 2 --------------------
+
+template <typename T>
+__global__ void tensor_max_grid_3channel_result_hip(float *srcPtr,
+                                                    uint xBufferLength,
+                                                    T *dstPtr)
+{
+    int id_x = hipThreadIdx_x * 8;
+    int id_z = hipBlockIdx_z;
+
+    __shared__ float partialRMax_smem[256];                           // 1024 floats of src reduced to 256 in a 256 x 1 thread block
+    __shared__ float partialGMax_smem[256];                           // 1024 floats of src reduced to 256 in a 256 x 1 thread block
+    __shared__ float partialBMax_smem[256];                           // 1024 floats of src reduced to 256 in a 256 x 1 thread block
+
+    uint srcIdx = (id_z * xBufferLength) * 3;
+    partialRMax_smem[hipThreadIdx_x] = srcPtr[srcIdx];                       // initialization of LDS for R channel to start of R channel using all 256 x 1 threads
+    partialGMax_smem[hipThreadIdx_x] = srcPtr[srcIdx + 1];                   // initialization of LDS for G channel to start of G channel using all 256 x 1 threads
+    partialBMax_smem[hipThreadIdx_x] = srcPtr[srcIdx + 2];                   // initialization of LDS for B channel to start of B channel using all 256 x 1 threads
+
+    if (id_x >= xBufferLength)
+        return;
+
+    int xAlignedLength = xBufferLength & ~7;                        // alignedLength for vectorized global loads
+    int xDiff = xBufferLength - xAlignedLength;                     // difference between bufferLength and alignedLength
+    srcIdx += id_x * 3;
+
+    if (id_x + 8 > xBufferLength)
+        srcIdx -= ((8 - xDiff) * 3);
```

Comment:
I don't understand this logic. srcIdx if a multiple of 3 and why the condition is checking 8 byte alignment?

Reply:
clarified